internal/receiver/scriptedinputsreceiver/procpipe/commander.go

```diff
@@ -0,0 +1,203 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procpipe
+
+import (
+	"context"
+	_ "embed"
+	"io"
+	"os/exec"
+	"strings"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+const includePattern = ". \"$(dirname \"$0\")\"/common.sh"
+
+//go:embed scripts/cpu.sh
```

Comment:
Since this isn't for windows I think there should be something like a `scripts_windows.go` with `//go:build windows
` and `script_others.go` with existence checks.